centrosome/zernike.py

```diff
@@ -85,21 +113,27 @@ def score_zernike(zf, radii, labels, indexes=None):
         indexes = np.arange(1,np.max(labels)+1,dtype=np.int32)
     else:
         indexes = np.array(indexes, dtype=np.int32)
-    radii = np.array(radii)
+    radii = np.asarray(radii, dtype=float)
+    n = radii.size
     k = zf.shape[2]
-    n = np.product(radii.shape)
     score = np.zeros((n,k))
     if n == 0:
         return score
-    areas = radii**2 * np.pi
+    areas = np.square(radii)
+    areas *= np.pi
```

Comment:
This is attempting to minimize the number of intermediate arrays numpy needs to allocate.  In this version of the code, `np.square(radii)` allocates memory for the result, assigned to `areas`. Then `areas *= np.pi` performs operation in place.

With `np.square(radii) * np.pi`, one temporary is allocated for output of `np.radii`, then another is allocated for the result of the product.

Evaluation of `areas = np.square(radii) * np.pi` also amounts to two calls, one for `square` and one for `multiply`, but uses an superfluous intermediate array for store the result of squaring.

Granted this style of writing is not helping readability, but it helps performance, especially for larger size inputs.